graph/patterns/change-tracking.md

```diff
@@ -0,0 +1,161 @@
+  2. [Optionally] Further GET requests to retrieve more pages of the current state via the `@odata.nextLink` URL.
+  3. After some time, a GET request to see if there are new changes via the `@odata.deltaLink`URL.
+  4. [Optionally] GET requests to retrieve more pages of changes via the `@odata.nextLink` URL.
+
+The `nextLink` provides a mechanism to do server-driven paging through the change data that is currently available.  When there are no further pages of changes immediately available, a `deltaLink` is returned instead.
+
+The `deltaLink` provides a mechanism for the API consumer to catch up on changes since their last request to the delta function, `deltaLink` or last page’s `nextLink`. If no changes have happened since the last request, then the deltaLink MUST return an empty collection.
+
+Both `nextLink` and `deltaLink` MUST be considered opaque URLs. The best practice is to make them opaque via encoding.
+
+Delta payload requirements
+  - The payload is a collection of change records using the collection format.
+  - The change records are full or partial representations of the resources according to their resource types.
+  - When a change representing a resource update is included in the payload the API producer MAY return either the changed properties or the full entity. The ID of the resource MUST be included in every change record.
+  - When an entity is deleted, the delta function MUST return the ID of the deleted entity as well as an `@removed` annotation with the reason field.
+  - When an entity is deleted, the reason MUST be set to “changed” if the entity can be restored.
+  - When an entity is deleted. the reason MUST be set to “deleted” if the entity cannot be restored.
+  - There is no mechanism to indicate that a resource has entered or exited the dataset based on a change that causes it to match or no longer match any `$filter` query parameter.
+  - When a link to an entity is deleted, when the linked entity is deleted, or when a link to an entity is added, the implementer MUST return a `property@delta` annotation. 
+    - When a link to an entity is deleted, but the entity still exists, the reason MUST be set to `changed`.
+    - When a link to an entity is deleted along with the entity, the reason MUST be set to `deleted`.
+  
+API producers MAY choose to collate multiple changes to the same resource into a single change record. 
+
+## When to use this pattern
+
+API consumers want a pull mechanism to request and process change to Microsoft Graph data, either via proactive polling or by responding to Microsoft Graph notifications.
+
+API consumers need guaranteed data integrity over the set of changes to Microsoft Graph data.
+
+## Issues and considerations
+
+ - API service MAY be able to respond to standard OData query parameters with the initial call to the delta function:
+
+    - `$select` to enforce the set of properties on which change is reported.
+    - `$filter` to influence the range of changes returned.
```

Comment:
"range" or "scope"?  Range makes it sound like a duration range, but we have other supported filters that are not duration based.